internal/pkg/manifest/validate.go

```diff
@@ -2050,29 +2054,38 @@ func validateDepsForEssentialContainers(deps map[string]containerDependency) err
 	return nil
 }
 
+type containerNameAndProtocol struct {
+	containerName     string
+	containerProtocol string
+}
+
 func validateExposedPorts(opts validateExposedPortsOpts) error {
-	containerNameFor := make(map[uint16]string)
-	populateMainContainerPort(containerNameFor, opts)
-	if err := populateSidecarContainerPortsAndValidate(containerNameFor, opts); err != nil {
+	portExposedTo := make(map[uint16]containerNameAndProtocol)
+
+	if err := populateAndValidateSidecarContainerPorts(portExposedTo, opts); err != nil {
 		return err
 	}
-	if err := populateALBPortsAndValidate(containerNameFor, opts); err != nil {
+	if err := populateAndValidateALBPorts(portExposedTo, opts); err != nil {
 		return err
 	}
-	if err := populateNLBPortsAndValidate(containerNameFor, opts); err != nil {
+	if err := populateAndValidateNLBPorts(portExposedTo, opts); err != nil {
+		return err
+	}
+	if err := populateAndValidateMainContainerPort(portExposedTo, opts); err != nil {
```

Comment:
I also had a question. Does the order matter of validation matter over here. 
I mean should we validate `MainContainer` followed by other validations.

Reply:
I think this was a product of iterative changes. As of right now I'm pretty sure the only thing that matters is that the sidecars are validated before the ALB/NLB. I can move this back to the front to make more sense.

Reply:
Just kidding! Today I remembered why that was! It's because in the event that you want to use different protocol on a port for the main container, the NLB/ALB should take priority in specifying the protocol!